runtime/src/bank.rs

```diff
@@ -1748,12 +1770,18 @@ impl Bank {
         let mut partitions = vec![];
         if parent_cycle < current_cycle {
             if current_cycle_index > 0 {
+                // generate and push gapped partitions because some slots are skipped
                 let parent_last_cycle_index = slot_count_in_two_day - 1;
+
+                // ... for parent cycle
                 partitions.push((
                     parent_cycle_index,
                     parent_last_cycle_index,
                     slot_count_in_two_day,
                 ));
+
+                // ... for current cycle
+                partitions.push((0, 0, slot_count_in_two_day));
```

Comment:
This fn (`fixed_cycle_partitions`) is mostly for benches, so doesn't need any gating like this: https://github.com/solana-labs/solana/pull/10206/files#diff-a7549f152920d85fb44e6a784b8e5e1fR1830